src/coreclr/src/vm/dynamicinterfacecastable.cpp

```diff
@@ -8,25 +8,51 @@
 
 namespace
 {
-    OBJECTREF CallGetInterfaceImplementation(OBJECTREF *objPROTECTED, const TypeHandle &interfaceTypeHandle, BOOL isinstTest)
+    BOOL CallIsInterfaceImplementated(OBJECTREF *objPROTECTED, const TypeHandle &interfaceTypeHandle, BOOL isDirectCast)
+    {
+        CONTRACT(BOOL) {
+            THROWS;
+            GC_TRIGGERS;
+            MODE_COOPERATIVE;
+            PRECONDITION(objPROTECTED != NULL);
+            PRECONDITION(interfaceTypeHandle.IsInterface());
+            POSTCONDITION(!isDirectCast || RETVAL);
+        } CONTRACT_END;
+
+        PREPARE_NONVIRTUAL_CALLSITE(METHOD__DYNAMICINTERFACECASTABLEHELPERS__IS_INTERFACE_IMPLEMENTED);
+
+        OBJECTREF managedType = interfaceTypeHandle.GetManagedClassObject(); // GC triggers
+
+        DECLARE_ARGHOLDER_ARRAY(args, 3);
+        args[ARGNUM_0] = OBJECTREF_TO_ARGHOLDER(*objPROTECTED);
+        args[ARGNUM_1] = OBJECTREF_TO_ARGHOLDER(managedType);
+        args[ARGNUM_2] = BOOL_TO_ARGHOLDER(isDirectCast);
+
+        BOOL isImplemented;
+        CALL_MANAGED_METHOD(isImplemented, BOOL, args);
```

Comment:
```suggestion
        CALL_MANAGED_METHOD(isImplemented, CLR_BOOL, args);
```
BOOL may have upped 3 bytes undefined on some platforms.